.github/workflows/ci-dgraph-tests.yml

```diff
@@ -0,0 +1,65 @@
+name: ci-dgraph-tests
+on:
+  push:
+    branches:
+      - main
+jobs:
+  dgraph-tests:
+    runs-on: [self-hosted, x64]
+    steps:
+      - name: Checkout Dgraph repo
+        uses: actions/checkout@v3
+        with:
+          repository: dgraph-io/dgraph
+          ref: main
+      - name: Get Go Version
+        run: |
+          #!/bin/bash
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; })
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Fetch latest Badger version
+        run: |
+          go get github.com/dgraph-io/badger/v4@main
```

Comment:
Trick is here: we are updating dep in dgraph to use latest version of badger on main branch